File: common/src/main/java/com/hartwig/actin/treatment/input/single/OneGeneTwoIntegers.java
package com.hartwig.actin.treatment.input.single;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class })
public abstract class OneGeneTwoIntegers {

    @NotNull
    public abstract String geneName();

    public abstract int integer1();

    public abstract int integer2();
}

File: common/src/main/java/com/hartwig/actin/treatment/input/single/OneGeneTwoIntegers.kt
package com.hartwig.actin.treatment.input.single

import org.immutables.value.Value
import org.jetbrains.annotations.NotNull

@Value.Immutable
@Value.Style(passAnnotations = [NotNull::class])
abstract class OneGeneTwoIntegers() {
    abstract fun geneName(): String
    abstract fun integer1(): Int
    abstract fun integer2(): Int
}
